### ISSUE TITLE
Re:  Problem with importing torrents if you've moved files in the uTorrent client.

### ISSUE THREAD

**Issue by OrangeFlavored** [Laharah/deluge-uTorrentImport#18]
Same problem here, am on Windows 10 64-bit with uTorrent 2.2.1.  Can give more info from my resume.dat if required, currently 200+ torrents fail to import properly due to this reason.  Some of them had individual files moved/renamed from the file list, while others had the whole directory moved from overall torrent list.  All files seem to revert to their default names rather than keeping new names as well, became a problem for stuff I was cross-seeding and had copied/renamed a few files that had differences between trackers.

**Comment by Laharah** [comment]
Can you give me some info about the torrents that failed to import?

The files that were moved, were they moved separate from the rest of the files? For instance, all the files were originally in a season folder and you moved one episode to another season? Or were these simply re-names?

**Comment by OrangeFlavored** [comment]
Ok I took a few samples out of my resume.dat and exported with Bencode editor.  Sent you in email since I might have forgotten to remove something and don't want to leak something accidentally like my user hash or w/e.

**Comment by Laharah** [comment]
I'll take a look at these when I get home and get back to you when I can.

**Comment by Laharah** [comment]
I can't see anything in these excerpts that should prevent them from being imported. There are only a few things I can think of. I'm assuming that the .torrent files existed on your machine?

second the files were definitely in the locations listed either under `'path'` or `'targets'`? Is the user that deluge is running under allowed to read/modify at the locations where the files were? No permission issues? Has the drive letter changed?

**Comment by OrangeFlavored** [comment]
Yeah, I double checked and the files were still definitely where those files indicated.  Didn't see any strange permissions and rerunning the plugin as admin didn't seem to help.  I do have some other torrents where the path was incorrect (had the torrent title appended to it as a nonexistent subdirectory) but the target had the correct path, but none of these ones I sent you seem to have that problem.

I have sometimes in the past relocated masses of torrents to a different drive/parent folder and then used BEncode Editor to fix all the paths at once, that wouldn't cause something like this would it?  I'd think if the paths were good enough for uTorrent it wouldn't matter though.

**Comment by Laharah** [comment]
well, the ebooks may have been changed in calibre. Did you ever re-check them in uTorrent? I know that by default calibre places a bookmark file in any ebook that you open inside calibre. There's a setting or two that if you don't turn off will change the ebooks themselves:

![calibre bookmark settings](https://i.imgur.com/isBxWTI.png)

Though I'm not sure why that would prevent them from being imported, they would fail their re-check though...

**Comment by OrangeFlavored** [comment]
Yeah, all the books recheck fine in uT

**Comment by Laharah** [comment]
Are there any non-ascii characters in your save paths? There could be a unicode error somewhere in the pipe, but I didn't see any in the resume info you sent me, and I'm pretty sure I've stamped those out, or at least they should tell you when they fail.

I just created a torrent in uTorrent and went through the ebook process you mentioned and everything re-imported and relocated as expected. I'm not sure what else could be going wrong. If you can create a new torrent and re-name/re-locate it in a way that breaks it I might be able to replicate the problem on my end. Or if you could get those debug logs I might be able to take some more guesses.

**Comment by OrangeFlavored** [comment]
I have a few torrents with non-ascii save paths but I don't think those ones I sent you are any of them, and that would only account for a small handful and not the hundreds that fail to import.

I'll see about importing again in debug mode and sending the logs over though.

**Comment by Laharah** [comment]
Okay, I think I might be able to reproduce these errors now that I've got all the library information. I'm seeing that a bunch of the torrents that didn't get renamed correctly have file rename failures for file existing. I'm not entirely sure why, but I can start to look into it. I'll update here when I have more.
